### PR TITLE
Upstream merge + a couple bugfixes [URGENT OPTIMIZATIONS]

### PR DESCRIPTION
**Pull request by drexample**
## About The Pull Request

Brings us up to date with Eris and fixes a little performance issue

## Why It's Good For The Game

Lags less and more content

## Changelog
:cl:
fix: Upstream merge and performance update!
/:cl:

### CODE REVIEW

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -100,21 +105,29 @@
 #define COMSIG_ATTACKBY "attack_by"										//from /mob/ClickOn():
 #define COMSIG_APPVAL "apply_values"									//from /atom/refresh_upgrades(): (/src) Called to upgrade specific values
 #define COMSIG_ADDVAL "add_values" 										//from /atom/refresh_upgrades(): (/src) Called to add specific things to the /src, called before COMSIG_APPVAL
-#define COMSIG_REMOVE "uninstall"										//from  /obj/item/weapon/tool/attackby(): Called to remove an upgrade
+#define COMSIG_REMOVE "uninstall"	
+#define COMSIG_ITEM_DROPPED	"item_dropped"					//from  /obj/item/weapon/tool/attackby(): Called to remove an upgrade
```
Comment (EvilJackCarver) [inline code/__DEFINES/components.dm:109]: You moved the comment down a line

Comment (drexample) [reply code/__DEFINES/components.dm:109]: https://github.com/discordia-space/CEV-Eris/blob/fd64a4815da5aa0062b934786bf85eff4503a2af/code/__DEFINES/components.dm

Ain't me

File: code/game/objects/items/weapons/storage/misc.dm [hunk 2]
```diff
@@ -72,3 +72,47 @@
 	new /obj/item/weapon/storage/fancy/mre_cracker(src)
 	new /obj/item/weapon/reagent_containers/food/snacks/candy/mre(src)
 	new /obj/item/weapon/material/kitchen/utensil/spoon/mre(src)
+
+/obj/item/weapon/storage/ration_pack/ihr
+	icon = 'icons/obj/food.dmi'
+	icon_state = "ihr_closed"
+	name = "ironhammer ration pack"
+	desc = "Silvery plastic package, with the letters \"IHR\" pasted onto the front. Seems air tight, and vacuumed sealed. \
```
Comment (EvilJackCarver) [inline code/game/objects/items/weapons/storage/misc.dm:80]: Ironhammer references

File: code/game/objects/items/weapons/tools/hammer.dm [hunk 3]
```diff
@@ -152,3 +174,18 @@
 		playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
 		user.throw_at(target, get_dist(target, user), 1, user)
 		T.stop()
+
+/obj/item/weapon/tool/hammer/IH
+	name = "FS \"Ironhammer\" Breaching Hammer"
+	desc = "A modified sledgehammer produced by Frozen Star for Ironhammer forces. This tool can take down standard walls and if the user is strong enough, reinforced walls."
```
Comment (EvilJackCarver) [inline code/game/objects/items/weapons/tools/hammer.dm:180]: Ironhammer references

File: code/modules/clothing/head/armor.dm [hunk 4]
```diff
@@ -77,7 +77,7 @@
 	flags_inv = BLOCKHEADHAIR|HIDEEARS
 
 /obj/item/clothing/head/armor/helmet/technomancer
-	name = "technomancer helmet"
+	name = "insulated technomancer helmet"
```
Comment (EvilJackCarver) [inline code/modules/clothing/head/armor.dm:80]: Technomancer reference

File: code/modules/clothing/spacesuits/rig/suits/station.dm [hunk 5]
```diff
@@ -158,10 +162,65 @@ Advanced Voidsuit: Chief Engineer
...
-	name = "Advanced Magboots"
+	name = "advanced magboots"
 	desc = "Advanced magnetic boots that have a lighter magnetic pull, placing less burden on the wearer."
 	mag_slow = 1
 
+/***************************************
+Technomancer RIG
+***************************************/
+/obj/item/weapon/rig/techno
+	name = "technomancer suit control module"
```
Comment (EvilJackCarver) [inline code/modules/clothing/spacesuits/rig/suits/station.dm:173]: Technomancer reference

File: code/modules/clothing/spacesuits/syndi.dm [hunk 6]
```diff
@@ -2,8 +2,9 @@
 /obj/item/clothing/head/space/syndicate
 	name = "red space helmet"
 	icon_state = "syndicate"
-	item_state = "syndicate"
-	desc = "A crimson helmet sporting clean lines and durable plating. Engineered to look menacing."
+	item_state = "syndi_helm"	//not sure why this isn't working, will try to fix before merge, but looks like a deep issue
+	desc = "A clasic crimson helmet sporting clean lines and durable plating."
```
Comment (EvilJackCarver) [inline code/modules/clothing/spacesuits/syndi.dm:6]: Classic (two S)

File: code/modules/clothing/spacesuits/syndi.dm [hunk 7]
```diff
@@ -19,7 +20,7 @@
 	name = "red space suit"
 	icon_state = "syndicate"
 	item_state = "space_suit_syndicate"
-	desc = "A crimson spacesuit sporting clean lines and durable plating. Robust, reliable, and slightly suspicious."
+	desc = "A clasic crimson spacesuit sporting clean lines and durable plating."
```
Comment (EvilJackCarver) [inline code/modules/clothing/spacesuits/syndi.dm:23]: Classic (two S)

File: code/modules/clothing/suits/armor.dm [hunk 8]
```diff
@@ -424,20 +424,20 @@
 	name = "full heavy armor vest"
 	desc = "A high-quality armor vest in a fetching tan. This one is webbed, and has kneepads and shoulderpads for extra coverage."
 	icon_state = "mercwebvest_fullbody"
+	slowdown = 0.15
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS
-
 //Technomancer armor
 /obj/item/clothing/suit/storage/vest/insulated
-	name = "insulated armor"
+	name = "insulated technomancer armor"
```
Comment (EvilJackCarver) [inline code/modules/clothing/suits/armor.dm:431]: Technomancer reference

File: code/modules/dungeons/junk_tractor_beam.dm [hunk 9]
```diff
@@ -0,0 +1,918 @@
...
+// Junk Field datum
+//////////////////////////////
+/datum/junk_field
+	var/name // name of the junk field
+	var/asteroid_belt_status // if it has an asteroid belt
+	var/affinity // affinity of the junk field
+	var/list/affinities = list(
+		"Neutral" = 10,
+		"OneStar" = 3,
+		"IronHammer" = 3,
```
Comment (EvilJackCarver) [inline code/modules/dungeons/junk_tractor_beam.dm:21]: Ironhammer reference. Fixing this is gonna be a pain, so just be aware of this one.

File: code/modules/materials/material_sheets.dm [hunk 10]
```diff
@@ -312,7 +312,7 @@
 
 /obj/item/stack/material/glass/phoronglass
 	name = "borosilicate glass"
-	desc = "This sheet is special platinum-glass alloy designed to withstand large temperatures"
+	desc = "This sheet is special plasma-glass alloy designed to withstand large temperatures"
```
Comment (EvilJackCarver) [inline code/modules/materials/material_sheets.dm:315]: Reference to plasma (versus phoron)

File: code/modules/materials/material_sheets.dm [hunk 11]
```diff
@@ -326,7 +326,7 @@
 
 /obj/item/stack/material/glass/phoronrglass
 	name = "reinforced borosilicate glass"
-	desc = "This sheet is special platinum-glass alloy designed to withstand large temperatures. It is reinforced with few rods."
+	desc = "This sheet is special plasma-glass alloy designed to withstand large temperatures. It is reinforced with few rods."
```
Comment (EvilJackCarver) [inline code/modules/materials/material_sheets.dm:329]: Reference to plasma

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 12]
```diff
@@ -262,3 +262,112 @@
...
 	..()
+
+// Runtime cat
+
+var/cat_cooldown = 20 SECONDS
+var/cat_max_number = 10
+var/cat_teleport = 0.0
+var/cat_number = 0
+
+/mob/living/simple_animal/cat/runtime
```
Comment (EvilJackCarver) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:273]: Ambiguous mob define, see line 231

File: code/modules/mob/living/simple_animal/hostile/syndicate.dm [hunk 13]
```diff
@@ -152,6 +152,8 @@
 	max_n2 = 0
 	minbodytemp = 0
 
+/mob/living/simple_animal/hostile/viscerator/emp_act(severity)
+	health -= 60*severity
```
Comment (EvilJackCarver) [inline code/modules/mob/living/simple_animal/hostile/syndicate.dm:156]: Inconsistent behaviour - lower EMP `severity` numbers are more severe.

File: code/modules/projectiles/guns/energy/stun.dm [hunk 14]
```diff
@@ -37,3 +37,9 @@
 	suitable_cell = /obj/item/weapon/cell/small
 	cell_type = /obj/item/weapon/cell/small
 	projectile_type = /obj/item/projectile/energy/electrode
+
+/obj/item/weapon/gun/energy/stunrevolver/moebius
+	name = "Moebius SP \"Zeus\""	//Should get it's own name
+	desc = "Also know as stunrevolver. A Moebius copy of the older and less precise Nanotrasen solution for non-lethal takedowns. This gun has smaller capacity in exchange for S-cells use."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/energy/stun.dm:43]: Moebius reference

File: code/modules/projectiles/guns/mods/research_mods.dm [hunk 15]
```diff
@@ -104,7 +104,7 @@
 // Add radiation damage to your weapon
 /obj/item/weapon/gun_upgrade/barrel/isotope_diffuser
 	name = "Moebius \"Atomik\" isotope diffuser"
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/mods/research_mods.dm:106]: Moebius reference

File: code/modules/projectiles/guns/projectile/automatic/c20r.dm [hunk 16]
```diff
@@ -46,3 +46,13 @@
 /obj/item/weapon/gun/projectile/automatic/c20r/Initialize()
 	. = ..()
 	update_icon()
+
+/obj/item/weapon/gun/projectile/automatic/c20r/moebius
+	name = "C-20M"
+	desc = "The C-20M is a Moebius copy of the famous C-20r, a lightweight and robust bullpup SMG of ancient design. \
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/automatic/c20r.dm:52]: Moebius reference

File: code/modules/tips_and_tricks/gameplay.dm [hunk 17]
```diff
@@ -8,6 +8,9 @@
 /tipsAndTricks/gameplay/fightBlob
     tipText = "If you're trying to take on a blob, a flamethrower or welder is your best weapon."
 
+/tipsAndTricks/gameplay/dismantleGun
+    tipText = "Did you know you can dismantle guns by using a wrench on them with harm intent ? Dismantling guns gives you gun parts."
```
Comment (EvilJackCarver) [inline code/modules/tips_and_tricks/gameplay.dm:12]: Please remove the space before the question mark

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
Got the review done. Didn't look at maps or icon files, but here's everything else.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 18]
```diff
@@ -552,10 +552,16 @@ var/global/list/damage_icon_parts = list()
 			under_state += w_uniform.icon_state + WORN_UNDER + get_gender_icon_contained(gender)
 
 		else if(w_uniform.icon_override)
-			under_icon = w_uniform.icon_override
+			// Eclipse edit start - Gendered icon_overrides
+			if(gender == MALE)
+				under_icon = w_uniform.icon_override
+			else
+				under_icon = w_uniform.icon_override_female
+			// Eclipse edit start - Gendered icon_overrides
```
Comment (EvilJackCarver) [inline code/modules/mob/living/carbon/human/update_icons.dm:560]: Misleading comment

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 12 above)
Comment (drexample) [reply code/modules/mob/living/simple_animal/friendly/cat.dm:273]: It's a cat that spawns whenever a runtime happens. It's not actually ambiguous.

File: code/modules/dungeons/junk_tractor_beam.dm (hunk 9 above)
Comment (drexample) [reply code/modules/dungeons/junk_tractor_beam.dm:21]: affinities are purely internal, so not a problem

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 12 above)
Comment (EvilJackCarver) [reply code/modules/mob/living/simple_animal/friendly/cat.dm:273]: I'm talking about the define, not the name or behaviour. 

Runtime the Cat is `/mob/living/simple_animal/cat/fluff/Runtime`, and Dusty is `/mob/living/simple_animal/cat/runtime` - the only difference aside from case being the `/fluff/` part of the define.

File: code/game/machinery/alarm.dm [hunk 19]
```diff
@@ -1141,24 +1144,26 @@ FIRE ALARM
...
 
 	if(locate(/obj/fire) in loc)
 		alarm()
-	
-	var/hivemind = check_for_hivemind()		//Eclipse edit: Hivemind stuff interferes with the equipment. That's the handwave I'm using for this balancing change. ^Spitzer
-	if(hivemind)
-		alarm()
-	
+
+	if(world.time - last_hive_check > 150) //Checks every 15 seconds
```
[the remark this answers is not in the document]
Comment (EvilJackCarver) [reply code/game/machinery/alarm.dm:1161]: Resolved through DMs

File: code/modules/dungeons/junk_tractor_beam.dm (hunk 9 above)
Comment (EvilJackCarver) [reply code/modules/dungeons/junk_tractor_beam.dm:21]: Looked like something read the affinities down farther below, which is why I brought it up.

**Comment by EvilJackCarver** [comment]
Once the review's addressed, you're cleared to speedmerge after testing.

**Review by EvilJackCarver — APPROVED** [review]

**Pull request merged**